### ISSUE TITLE
(CSNG) Ignition pin enhancement

### ISSUE THREAD

**Issue by Maju3** [htruong/crankshaft#242]
If you need support: Please post on https://www.reddit.com/r/crankshaft/ as this is a bug tracker, not a support forum.

Note: Reporting issues not based on official published builds will be closed without any comment.

- What phone do you have?

- What OS version?

- What crankshaft version?

- Steps to reproduce the bug, if you're planning to report a bug. Please indicate whether the bug is always repoducible.

- Please provide any further information that you might find helpful if available.

- Starting with pre4: Boot into dev mode (can be selected in settings).
  Reproduce the possible bug. Create the debug logs by pushing the grey button on bottom of screen.
  Add the created file (/boot/debug.zip) to the issue.

  Warning: if you have configured wifi passwords they will be blanked by script during creation.
  We can't make absolutly sure you passwords are removed so do a personal check of the files inside the zip.

Okay just an idea:

For now it works like this:
5volts on the selected ignition gpio keeps the System running.
0volts on the selected ignition gpio FOR THE SELECTED DELAY TIME will initiate a shutdown.
This means the ignition pin delay time is to Keep the System running while you crank your cars Motor to avoid unwanted shutdown.
Now my idea:
Is it possible to use the ignition pin to trigger the "General shutdown" in cs base Settings?
I mean this shutdown timer which is triggered by disconnecting phone for now.
You just Need a Setting if this General shutdown timer should be triggered by disconnecting phone OR by ignition pin.

For the ignitionpin DELAY you could use a DEFAULT of 1 Minute.
Then the IGNITION pin should trigger the SCREEN BACKLIGHT first and the start the "General shutdown timer".
I try to explain it:

5volt on the ignition pin turns screen backlight ON.
0volts on ignition pin (FOR 1 MINUTE DEFAULT) turns screen backlight OFF and starts the General shutdown timer.

Example:
you set the GENERAL shutdown timer to 30 minutes.
Now you leave your car just for a short time of lets say 10 minutes.
You turn ignition off and after 1 Minute the screen backlight goes off.
This means no Attention on your car from thiefs.
The shutdown timer of 30 minutes runs but if you come back to your car before this time then you turn ignition on and the screen backlight is ON and you are Ready to go.
This means no Long boot times at short stops at the gas Station;)

So basically the IGNITION pin should turn on the screen backlight when ignition is ON.
When ignition is OFF for 1 Minute it should turn off the screen AND start the General shutdown timer.

What do you think?

**Comment by hawkeyexp** [comment]
I'm sorry - releasing one thing (extended delay now possible) you wan't x others which are partitial vise versa the reqeust before... why not disabling all stuff and releasing with your own script? There are some default cases defined in a clean way which could be used and working for most users. If you need complex setup's you have to release it in you're custom way.

**Comment by Maju3** [comment]
Sorry man.
I never said that I want this!
Thats why I write "JUST AN IDEA".
And thats why I write "What do you think?".

Anyway i think this would be a nice feature.
